components/dashboard/src/components/ItemsList.tsx

```diff
@@ -28,12 +28,18 @@ export function Item(props: {
 }
 
 export function ItemField(props: {
-    children?: React.ReactNode;
-    className?: string;
+  children?: React.ReactNode
+  className?: string
+  shouldCenterVertically?: boolean
 }) {
-    return <div className={`flex-grow my-auto mx-1 ${props.className || ""}`}>
-        {props.children}
-    </div>;
+  return (
+    <div className={`flex-grow mx-1 ${props.className || ""} ${props.shouldCenterVertically ? "my-auto" : "my-0"}`}>
```

Comment:
Thank you for making the vertical centering optional!

However, I see that you can now set the vertical margin via both the `className` (e.g. by setting `my-auto` and/or `my-0`) or the new `shouldCenterVertically` boolean. This creates a slight ambiguity, as future developers might accidentally provide conflicting margin classes, and then wonder why their vertical centering doesn't work.

How about shifting the responsibility of `my-auto` vs `my-0` (or no `my` class at all) to the users of `ItemField` instead? (I.e. remove `my-auto` from the className here as you did, but don't introduce a `props.shouldCenterVertically` -- simply let all the users of `ItemField` provide a `my-auto` via the `props.className` if needed).

Please let me know what you think of this alternative proposal. 🙂

Reply:
100% agree!

I don't know if you had the chance to take a look at my commit messages, but the gist of it is I chose to add a prop purposely for limiting the scope of my intervention and "don't go too far my first time around".

I'm totally up for a refactoring involving the 5-6 components where this one is used and removing the new prop, and I'll work on it as soon as I am able.

As a side note, there's also something that makes my spider-senses tingle about having to use both `my-auto` and `items-center` to center things, but at this moment I just don't know Tailwind nor how it's used in this codebase well enough to propose anything about it.

Reply:
Refactoring done!

Reply:
> Refactoring done!

Wow, many thanks! 💯 🙏 I'll take another look at your changes today.

> As a side note, there's also something that makes my spider-senses tingle about having to use both `my-auto` and `items-center` to center things, but at this moment I just don't know Tailwind nor how it's used in this codebase well enough to propose anything about it.

You're absolutely right, that's an excellent point. 🎯 From my perspective:
- centering things with flex is far superior to centering with margins (flex makes it quite easy, and the produced layout "makes sense" and works well in the majority of situations regardless of screen size, doesn't create awkward overflows, etc.)
- there is still a lot of margin-centering in the code base (or, worse, both margin- and flex-centering combined 🙈) which could be improved by removing all the auto margins and using flex instead

However, in the interest of not derailing this Pull Request too much, maybe let's focus on the `my-auto` refactoring here. We can always switch to flex in future Pull Requests. 😁